processing/src/test/java/org/apache/druid/segment/projections/ProjectionsTest.java

```diff
@@ -0,0 +1,297 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.segment.projections;
+
+import org.apache.druid.data.input.impl.AggregateProjectionSpec;
+import org.apache.druid.data.input.impl.LongDimensionSchema;
+import org.apache.druid.data.input.impl.StringDimensionSchema;
+import org.apache.druid.query.aggregation.LongSumAggregatorFactory;
+import org.apache.druid.query.filter.EqualityFilter;
+import org.apache.druid.query.filter.Filter;
+import org.apache.druid.query.filter.LikeDimFilter;
+import org.apache.druid.segment.AggregateProjectionMetadata;
+import org.apache.druid.segment.CursorBuildSpec;
+import org.apache.druid.segment.column.ColumnType;
+import org.apache.druid.segment.column.RowSignature;
+import org.apache.druid.segment.filter.AndFilter;
+import org.apache.druid.segment.filter.IsBooleanFilter;
+import org.apache.druid.segment.filter.OrFilter;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+class ProjectionsTest
+{
+  @Test
+  void testSchemaMatchSimple()
+  {
+    RowSignature baseTable = RowSignature.builder()
+                                         .addTimeColumn()
+                                         .add("a", ColumnType.LONG)
+                                         .add("b", ColumnType.STRING)
+                                         .add("c", ColumnType.LONG)
+                                         .build();
+    AggregateProjectionMetadata spec = new AggregateProjectionMetadata(
+        AggregateProjectionSpec.builder("some_projection")
+                               .groupingColumns(new LongDimensionSchema("a"), new StringDimensionSchema("b"))
+                               .aggregators(new LongSumAggregatorFactory("c_sum", "c"))
+                               .build()
+                               .toMetadataSchema(),
+        12345
+    );
+    CursorBuildSpec cursorBuildSpec = CursorBuildSpec.builder()
+                                                     .setPreferredOrdering(List.of())
+                                                     .setAggregators(
+                                                         List.of(
+                                                             new LongSumAggregatorFactory("c", "c")
+                                                         )
+                                                     )
+                                                     .build();
+
+    Projections.ProjectionMatch projectionMatch = Projections.matchAggregateProjection(
+        spec.getSchema(),
+        cursorBuildSpec,
+        new RowSignatureChecker(baseTable)
+    );
+    Projections.ProjectionMatch expected = new Projections.ProjectionMatch(
```

Comment:
nit- moving `ProjectionMatch` out into its own class would help a little with making the test code more concise.